test/jit/test_backends.py

```diff
@@ -685,3 +687,75 @@ def test_execution(self):
     @skipIfRocm
     def test_errors(self):
         self.error_module_compiler_test.test_errors()
+
+"""
+Unit Tests for Nnapi backend with delegate
+"""
+# This is needed for IS_WINDOWS or IS_MACOS to skip the tests.
+@unittest.skipIf(TEST_WITH_ROCM or IS_SANDCASTLE or IS_WINDOWS or IS_MACOS or IS_FBCODE,
+                 "Non-portable load_library call used in test")
+class NnapiBackendPReLUTest(JitTestCase):
+    """
+    Test lowering a simple PRelU module to Nnapi backend.
+    TODO: After Nnapi delegate is finished T91991928,
+    add tests for running the module (currently only lowers the model)
+    """
+    def setUp(self):
+        super().setUp()
+        # Load nnapi delegate library
+        torch_root = Path(__file__).resolve().parent.parent.parent
+        p = torch_root / 'build' / 'lib' / 'libnnapi_backend.so'
+        torch.ops.load_library(str(p))
+
+        # Nnapi set up
+        # Avoid saturation in fbgemm
+        torch.backends.quantized.engine = 'qnnpack'
+
+        libneuralnetworks_path = os.environ.get("LIBNEURALNETWORKS_PATH")
+        if libneuralnetworks_path:
+            ctypes.cdll.LoadLibrary(libneuralnetworks_path)
+            print("Will attempt to run NNAPI models.")
+            self.can_run_nnapi = True
+        else:
+            self.can_run_nnapi = False
+
+    def test_execution(self):
+        # Save default dtype
+        module = torch.nn.PReLU()
+        default_dtype = module.weight.dtype
+        # Change dtype to float32 (since float 64 is not supported)
```

Comment:
This is odd, can you provide more context on why is this needed (which still sounds worrisome as a workaround).

Reply:
It seems that the default dtype has been changed to float64 for this file. The problem seems specific to test_backends.py, since when I wrote a local script, I didn't need to do this. 
Since dtype is global, I think some other test is setting the dtype to float64. For example Akshit found this one, https://www.internalfb.com/code/fbsource/[a46774f32bd8]/fbcode/caffe2/test/jit/test_complexity.py?lines=8.
I was also running into problems with other tests after setting dtype to float32, so I had to add code to change it back to float64.

Reply:
@Amyh11325 mind setting it back after the test in test_complexity and seeing if we still need it?

Reply:
I can try, but I think it would break other tests that require the default dtype to be float64, like when I didn't change dtype back to float64 in this PR.